lib/postgres_to_redshift/keys.rb

```diff
@@ -0,0 +1,35 @@
+module PostgresToRedshift
+  class Keys
+    def initialize(source_connection:)
+      @source_connection = source_connection
+    end
+
+    def all
+      all_keys
+    end
+
+    private
+
+    attr_reader :source_connection
+
+    def all_keys
+      source_connection.exec(find_all_keys_sql).map do |key_attributes|
+        Key.new(attributes: key_attributes)
+      end
+    end
+
+    def find_all_keys_sql
+      <<~SQL
+        select
+          conrelid::regclass AS table_name,
+          conname as name,
+          pg_get_constraintdef(oid) as key_definition,
+          contype as key_type
+        from pg_constraint
+        where contype IN ('f', 'p ')
+        and connamespace = 'public'::regnamespace
```

Comment:
Should this be limited to constraints relating to just those tables specified in the import (both sides of the constraint for the FKs)?

Reply:
Ah good point, let me do that.